.babelrc

```diff
@@ -0,0 +1,3 @@
+{
+	"presets": ["@babel/preset-env"]
+}
```

Comment:
Do you agree on this preset as solution to transpile ES Modules when running `pika`?

Reply:
In `@pika/plugin-build-node`, the option to use custom babelrc files is disabled.This wouldn't have an effect
https://github.com/pikapkg/builders/blob/master/packages/plugin-build-node/src/index.ts#L47

Reply:
> In `@pika/plugin-build-node`, the option to use custom babelrc files is disabled.This wouldn't have an effect
> [pikapkg/builders:packages/plugin-build-node/src/index.ts@`master`#L47](https://github.com/pikapkg/builders/blob/master/packages/plugin-build-node/src/index.ts#L47) [](/pikapkg/builders/blob/HEAD@{2020-04-16T21:43:21Z}/packages/plugin-build-node/src/index.ts#L47)

Then... I would say to *not* migrate to ES Modules on this PR so we are less aggressive on these changes and in a new PR/Issue it can be discussed how to tackle this (if it's interesting to cover). What do you think? @wolfy1339

Reply:
Convert all the files to TypeScript and keep using the same `import`/`export` style as you are already doing.

We could make our own plugin in the meantime until pika releases their own ESM builder, best to leave that for another PR though.

Reply:
> Convert all the files to TypeScript and keep using the same `import`/`export` style as you are already doing.
> 
> We could make our own plugin in the meantime until pika releases their own ESM builder, best to leave that for another PR though.

Sounds good to me :)

Reply:
> Do we still need this now that the source code is written in TypeScript?

If we don't use it the transpilation of `import / export` is not done.

Options that came to my mind:
* keep .babelrc and PR to pika support for import/export or create our own plugin as @wolfy1339 pointed
* transpile it without pika for the moment

### Example pika/dist-src/index.js

<details>
 <summary>*with .babelrc*</summary>
<code>
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createWebhooksApi = createWebhooksApi;
Object.defineProperty(exports, "createEventHandler", {
  enumerable: true,
  get: function get() {
    return _index.createEventHandler;
  }
});
Object.defineProperty(exports, "createMiddleware", {
  enumerable: true,
  get: function get() {
    return _index2.createMiddleware;
  }
});
Object.defineProperty(exports, "sign", {
  enumerable: true,
  get: function get() {
    return _index3.sign;
  }
});
Object.defineProperty(exports, "verify", {
  enumerable: true,
  get: function get() {
    return _index4.verify;
  }
});

var _index = require("./event-handler/index.js");

var _index2 = require("./middleware/index.js");

var _middleware = require("./middleware/middleware.js");

var _index3 = require("./sign/index.js");

var _index4 = require("./verify/index.js");

var _verifyAndReceive = require("./middleware/verify-and-receive.js");

function createWebhooksApi(options) {
  if (!options || !options.secret) {
    throw new Error("options.secret required");
  }

  var state = {
    eventHandler: (0, _index.createEventHandler)(options),
    path: options.path || "/",
    secret: options.secret,
    hooks: {}
  };
  return {
    sign: _index3.sign.bind(null, options.secret),
    verify: _index4.verify.bind(null, options.secret),
    on: state.eventHandler.on,
    removeListener: state.eventHandler.removeListener,
    receive: state.eventHandler.receive,
    middleware: _middleware.middleware.bind(null, state),
    verifyAndReceive: _verifyAndReceive.verifyAndReceive.bind(null, state)
  };
}
</code>
</details>

<details>
 <summary>*without .babelrc*</summary>
<code>
import { createEventHandler } from "./event-handler/index.js";
import { createMiddleware } from "./middleware/index.js";
import { middleware } from "./middleware/middleware.js";
import { sign } from "./sign/index.js";
import { verify } from "./verify/index.js";
import { verifyAndReceive } from "./middleware/verify-and-receive.js";

function createWebhooksApi(options) {
  if (!options || !options.secret) {
    throw new Error("options.secret required");
  }

  const state = {
    eventHandler: createEventHandler(options),
    path: options.path || "/",
    secret: options.secret,
    hooks: {}
  };
  return {
    sign: sign.bind(null, options.secret),
    verify: verify.bind(null, options.secret),
    on: state.eventHandler.on,
    removeListener: state.eventHandler.removeListener,
    receive: state.eventHandler.receive,
    middleware: middleware.bind(null, state),
    verifyAndReceive: verifyAndReceive.bind(null, state)
  };
}

export { createEventHandler, createMiddleware, createWebhooksApi, sign, verify };
```
</code>
</details>

Reply:
I'm sorry I'm not following ... it's okay if the `pkg/dist-src/*` files use import/export. That's what I have in the other packages, too. For example: https://unpkg.com/browse/@octokit/rest@17.7.0/dist-src/index.js

The `dist-src` folder is not what is used by dependants, they either use `dist-web/` or `dist-node/`

Reply:
> I'm sorry I'm not following ... it's okay if the `pkg/dist-src/*` files use import/export. That's what I have in the other packages, too. For example: [unpkg.com/browse/@octokit/rest@17.7.0/dist-src/index.js](https://unpkg.com/browse/@octokit/rest@17.7.0/dist-src/index.js)
> 
> The `dist-src` folder is not what is used by dependants, they either use `dist-web/` or `dist-node/`

It's because of what you pointed [here](https://github.com/octokit/webhooks.js/pull/113#discussion_r420313771).